### ISSUE TITLE
CharTensor & ShortTensor not handled correctly

### ISSUE THREAD

**Issue by d11** [deepmind/torch-hdf5#6]

**Comment by d11** [comment]
NB use https://www.hdfgroup.org/HDF5/doc/RM/RM_H5T.html#Datatype-GetSign

**Comment by vadimkantorov** [comment]
bump... my use case: hdf5 torch object de/serialization; transforming automatically CharTensor's to strings after deserialization.

**Comment by d11** [comment]
Would a workaround be to go via ByteTensor, perhaps?

**Comment by nhynes** [comment]
@vadimkantorov I just posted a pull request for decoding the HDF5 String type to CharTensors. Does this at least somewhat solve your problem?

Regarding automatic strings, this would be nice for your [and my] use case but it's ambiguous whether the user wants a signed byte or an ascii character.

**Comment by vadimkantorov** [comment]
Yes, I guess it would. And I think it's nice to have mapping between native type and Torch tensor type to be done consistently (even regardless of serialization stuff).

Thinking more about it, a fix like this isn't enough for a full serialization solution (how do we store booleans or numbers? if we store them as 1-element tensors, do we automatically squeeze all read tensors?). Attribute support would be a better way to preserve type information.
